nix/services.nix

```diff
@@ -16,12 +16,24 @@
           port = 6381;
         };
         services.zookeeper."zookeeper1".enable = true;
-        services.apache-kafka."kafka1".enable = true;
+        services.apache-kafka."kafka1" = {
+          enable = true;
+          settings = {
+            # Since the available brokers are only 1
+            "offsets.topic.replication.factor" = 1;
+            "zookeeper.connect" = [ "localhost:2181" ];
```

Comment:
The first setting can probably be part of default settings in `services-flake` but that default will be inaccurate when multiple instances of kafka are used, by explicitly increasing the number of brokers in `settings`. 

It doesn’t make sense to have `zookeeper.connect` in default settings because in the recent versions of kafka they have a feature that can [remove the dependency on zookeeper completely](https://developer.confluent.io/learn/kraft/). I can avoid the hardcoding of `localhost:2181` by picking the values from zookeeper config.

Reply:
Unrelated, but created an issue to document using `kraft`: https://github.com/juspay/services-flake/issues/394